server/src/main/java/org/elasticsearch/search/fetch/FetchSubPhase.java

```diff
@@ -75,6 +77,13 @@ public IndexReader topLevelReader() {
             return cache;
         }
 
+        public void analyzer(Analyzer analyzer) {
+            this.analyzer=analyzer;
+        }
+        public Analyzer analyzer() {
+            return analyzer;
+        }
+
```

Comment:
I think we should just use the already-existing object cache to pass this analyzer around? It seems a bit odd to add a specialized object that is only used by a plugin.  We could also cache the parsed annotations rather than the analyzer itself, and store a reference to the HitContext on the analyzer

Reply:
We still need to cache the analyzer because the creation and the initialization of the analyzer occur in different methods of the highlighter and we need to pass this through

Reply:
But the initialization just parses the annotations.  So if instead you construct the Analyzer with a reference to the HitContext, and then cache the annotations on that same HitContext in `loadFieldValues()` then the Analyzer can access them without it sitting in the cache.